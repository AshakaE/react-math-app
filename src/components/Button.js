import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, clickHandler }) => (
  <button className="w-6" type="button" onClick={() => clickHandler(name)}>
    {name}
  </button>
);

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};
export default Button;
