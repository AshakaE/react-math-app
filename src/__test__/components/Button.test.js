import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../../components/Button';
import '@testing-library/jest-dom';

const test = () => {};

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button name="1" clickHandler={test} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with the given name', () => {
    render(<Button name="test" clickHandler={test} />);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('test');
  });
});
