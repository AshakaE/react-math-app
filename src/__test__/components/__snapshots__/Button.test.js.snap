// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="w-1/4 border border-gray-400 text-2xl focus:outline-none gray "
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
